core/data/src/main/java/com/nolawiworkineh/data/networking/HttpClientFactory.kt

```diff
@@ -0,0 +1,51 @@
+@file:OptIn(ExperimentalSerializationApi::class)
+
+package com.nolawiworkineh.data.networking
+
+import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
+import com.nolawiworkineh.core.data.BuildConfig
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.json.Json
+import okhttp3.Interceptor
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+
+
+class HttpClientFactory {
+
+    fun create(): Retrofit {
+        // Logging Interceptor
+        val loggingInterceptor = HttpLoggingInterceptor().apply {
+            level = HttpLoggingInterceptor.Level.BODY
+        }
+
+        // Header Interceptor for adding API key
+        val headerInterceptor = Interceptor { chain ->
+            val request = chain.request().newBuilder()
+                .addHeader("Authorization", "Bearer ${BuildConfig.API_KEY}")
```

Comment:
Prefer to use `.header("Authorization", "Bearer ${BuildConfig.API_KEY}”)` here. It acts like a database upsert operation and adds a new header and replaces any that already exist with the same name. Whereas `addHeader()` will add another header value with the same name and not replace existing ones.

Requests with multiple headers of the same name value can be problematic sometimes with the server that processes the request and this issue is hard to diagnose.